test/any/Diff-test.js

```diff
@@ -0,0 +1,961 @@
+      // Expecting left to be undefined and right to be defined.
+      expect(chunks[0].leftValue).toBeUndefined();
+      expect(chunks[0].rightValue).toBeDefined();
+    });
+
+    it('should return a diff fragment if one has a member and one does not', function() {
+      var leftSrc = 'TestSrc1';
+      var rightSrc = 'TestSrc2';
+      var interfaceName = 'Test';
+      var firstMap = createMap(`interface ${interfaceName} { serializer; };`, leftSrc);
+      var secondMap = createMap(`interface ${interfaceName} {};`, rightSrc);
+
+      var chunks = differ.diff(firstMap, secondMap);
+      expect(chunks.length).toBe(1);
+      // Expecting definitionName field to be name of interface.
+      expect(chunks[0].definitionName).toBe(interfaceName);
+      // Expecting status to represent no match.
+      expect(chunks[0].status).toBe(DiffStatus.NO_MATCH_ON_RIGHT);
+      // Expecting difference to be in members field of definition.
+      expect(chunks[0].propPath).toBe('.definition.members');
+      // Expecting left to be defined, and right undefined.
+      expect(chunks[0].leftValue).toBeDefined();
+      expect(chunks[0].rightValue).toBeUndefined();
+      // Expecting left source and right source to be correctly populated.
+      expect(chunks[0].leftSources).toBeDefined();
+      expect(chunks[0].leftSources.length).toBe(1);
+      expect(chunks[0].leftSources[0]).toBe(leftSrc);
+      expect(chunks[0].rightSources).toBeDefined();
+      expect(chunks[0].rightSources.length).toBe(1);
+      expect(chunks[0].rightSources[0]).toBe(rightSrc);
+    });
+
+    it('should return a diff fragment if members have different values', function() {
+      var firstMap = createMap(`interface Test { const boolean isPotato = true; };`);
+      var secondMap = createMap(`interface Test { const boolean isPotato = false; };`);
+
+      var chunks = differ.diff(firstMap, secondMap);
+      expect(chunks.length).toBe(1);
+      // Expecting difference to be in the value field of the member isPotato.
+      expect(chunks[0].propPath).toBe('.definition.members[isPotato].member.value.literal');
+      expect(chunks[0].status).toBe(DiffStatus.VALUES_DIFFER);
+      expect(chunks[0].leftValue).toBe('true');
+      expect(chunks[0].rightValue).toBe('false');
+    });
+
+    it('should return a diff fragment if members have different types', function() {
+      var firstMap = createMap(`interface Test { attribute boolean isPotato; };`);
+      var secondMap = createMap(`interface Test { attribute unsigned long isPotato; };`);
+
+      var chunks = differ.diff(firstMap, secondMap);
+      expect(chunks.length).toBe(1);
+      // Expecting difference to be in the type field of member.
+      expect(chunks[0].propPath).toBe('.definition.members[isPotato].member.type.name.literal');
+      expect(chunks[0].status).toBe(DiffStatus.VALUES_DIFFER);
+      expect(chunks[0].leftValue).toBe('boolean');
+      expect(chunks[0].rightValue).toBe('unsigned long');
```

Comment:
This comment is applies to types in general.
If one type is defined in the grammar (e.g. `long`) and the other is a custom type (e.g. `Session`), the diffChunk will be fairly vague and will not be as specific as the diff result here.